packages/big-design/src/components/Checkbox/spec.tsx

```diff
@@ -54,115 +56,102 @@ describe('render Checkbox', () => {
+    render(<Checkbox label="Checked" checked={false} onChange={() => null} disabled />);
 
-    expect(container.firstChild).toMatchSnapshot();
+    expect(screen.getByRole('checkbox').parentElement).toMatchSnapshot();
   });
 
   test('disabled indeterminate', () => {
-    const { container } = render(<Checkbox label="Unchecked" isIndeterminate={true} onChange={() => null} disabled />);
+    render(<Checkbox label="Unchecked" isIndeterminate={true} onChange={() => null} disabled />);
 
-    expect(container.firstChild).toMatchSnapshot();
+    expect(screen.getByRole('checkbox').parentElement).toMatchSnapshot();
   });
 });
 
 test('has correct value for checked', () => {
-  const { getByTestId } = render(
-    <Checkbox label="Checked" checked={true} onChange={() => null} data-testid="checkbox" />,
-  );
-  const input = getByTestId('checkbox') as HTMLInputElement;
+  render(<Checkbox label="Checked" checked={true} onChange={() => null} />);
 
-  expect(input.checked).toBe(true);
+  expect(screen.getByRole('checkbox')).toBeChecked();
 });
 
 test('has correct value for unchecked', () => {
-  const { getByTestId } = render(
-    <Checkbox label="Checked" checked={false} onChange={() => null} data-testid="checkbox" />,
-  );
-  const input = getByTestId('checkbox') as HTMLInputElement;
+  render(<Checkbox label="Checked" checked={false} onChange={() => null} />);
 
-  expect(input.checked).toBe(false);
+  expect(screen.getByRole('checkbox')).not.toBeChecked();
 });
 
 test('has correct value for indeterminate', () => {
-  const { getByTestId } = render(
-    <Checkbox label="Checked" isIndeterminate checked={false} onChange={() => null} data-testid="checkbox" />,
-  );
-  const input = getByTestId('checkbox') as HTMLInputElement;
+  render(<Checkbox label="Checked" isIndeterminate checked={false} onChange={() => null} />);
 
-  expect(input.checked).toBe(false);
+  expect(screen.getByRole('checkbox')).not.toBeChecked();
 });
 
 test('triggers onChange when clicking the checkbox', () => {
   const onChange = jest.fn();
-  const { getByTestId } = render(
-    <Checkbox label="Checked" checked={true} onChange={onChange} data-testid="checkbox" />,
-  );
-  const checkbox = getByTestId('checkbox') as HTMLInputElement;
 
-  fireEvent.click(checkbox);
+  render(<Checkbox label="Checked" checked={true} onChange={onChange} />);
+
+  userEvent.click(screen.getByRole('checkbox'));
 
   expect(onChange).toHaveBeenCalled();
 });
 
 test('triggers onChange when clicking styled and text label', () => {
   const onChange = jest.fn();
-  const { container } = render(<Checkbox label="Checked" checked={true} onChange={onChange} data-testid="checkbox" />);
 
-  const labels = container.querySelectorAll('label');
+  render(<Checkbox label="Checked" checked={true} onChange={onChange} />);
+
+  const labels = screen.getByRole<HTMLInputElement>('checkbox').parentElement!.querySelectorAll('label');
```

Comment:
Could this be `screen.getAllByRole('label')`?

Reply:
I was also looking on similar approach but looks like such role does not exist

Reply:
what about `getByText(/checked/)`? might still be better than all this... 🤷

Reply:
In this case we will not catch `label` with `svg` inside

Reply:
Totally forgot there is the `screen.getByLabelText('')` query 🤦

Reply:
Query above may return an input associated with the label, but we need to collect all labels and then iterate through them. Can you provide some code according to our component? Maybe I have wrong understanding of your idea